### ISSUE TITLE
can't install with python3.7

### ISSUE THREAD

**Issue by Gongxter** [lrq3000/unireedsolomon#8]
Hi, somehow i cant install this via pip. I've tried it via the git repo as well as directly via pip install unireedsolomon with either python 3.7 64 bit and 32 bit.
I installed python version 3.6 64 bit and it installs successfully.
Is there a problem with my python 3.7 version ? Or is there another way to get this working with python version 3.7?

**Comment by Gongxter** [comment]
So apparently after installing cython i could install it.

**Comment by lrq3000** [comment]
Hello, thank you for reporting this, normally Cython should not be necessary for using the pure python implementation, I'll have a look.

**Comment by CompSciMaj13** [comment]
Also was not able to install using python 3.7.5 and received the following error. I'm inside a pipenv virtual environment and had to install cython first before I was able to install unireedsolomon.

> tl;dr Workaround: Install cython first

```
gcc -pthread -Wno-unused-result -Wsign-compare -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D_GNU_SOURCE -fPIC -fwrapv -fPIC -I/usr/include/python3.7m -c unireedsolomon/cff.c -o build/temp.linux-x86_64-3.7/unireedsolomon/cff.o          
    unireedsolomon/cff.c: In function ‘__Pyx_ExceptionSave’:                                                                                                    
    unireedsolomon/cff.c:24548:21: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_type’; did you mean ‘curexc_type’?                        
    24548 |     *type = tstate->exc_type;                                                                                                                       
          |                     ^~~~~~~~                                                                                                                        
          |                     curexc_type                                                                                                                     
    unireedsolomon/cff.c:24549:22: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_value’; did you mean ‘curexc_value’?                      
    24549 |     *value = tstate->exc_value;                                                                                                                     
          |                      ^~~~~~~~~                                                                                                                      
          |                      curexc_value                                                                                                                   
    unireedsolomon/cff.c:24550:19: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_traceback’; did you mean ‘curexc_traceback’?              
    24550 |     *tb = tstate->exc_traceback;                                                                                                                    
          |                   ^~~~~~~~~~~~~                                                                                                                     
          |                   curexc_traceback                                                                                                                  
    unireedsolomon/cff.c: In function ‘__Pyx_ExceptionReset’:                                                                                                   
    unireedsolomon/cff.c:24562:24: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_type’; did you mean ‘curexc_type’?                        
    24562 |     tmp_type = tstate->exc_type;                                                                                                                    
          |                        ^~~~~~~~                                                                                                                     
          |                        curexc_type                                                                                                                  
    unireedsolomon/cff.c:24563:25: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_value’; did you mean ‘curexc_value’?                      
    24563 |     tmp_value = tstate->exc_value;                                                                                                                  
          |                         ^~~~~~~~~                                                                                                                   
          |                         curexc_value                                                                                                                
    unireedsolomon/cff.c:24564:22: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_traceback’; did you mean ‘curexc_traceback’?              
    24564 |     tmp_tb = tstate->exc_traceback;                                                                                                                 
          |                      ^~~~~~~~~~~~~                                                                                                                  
          |                      curexc_traceback                                                                                                               
    unireedsolomon/cff.c:24565:13: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_type’; did you mean ‘curexc_type’?                        
    24565 |     tstate->exc_type = type;                                                                                                                        
          |             ^~~~~~~~                                                                                                                                
          |             curexc_type                                                                                                                             
    unireedsolomon/cff.c:24566:13: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_value’; did you mean ‘curexc_value’?                      
    24566 |     tstate->exc_value = value;                                                                                                                      
          |             ^~~~~~~~~                                                                                                                               
          |             curexc_value                                                                                                                            
    unireedsolomon/cff.c:24567:13: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_traceback’; did you mean ‘curexc_traceback’?              
    24567 |     tstate->exc_traceback = tb;                                                                                                                     
          |             ^~~~~~~~~~~~~                                                                                                                           
          |             curexc_traceback                                                                                                                        
    unireedsolomon/cff.c: In function ‘__Pyx_GetException’:                                                                                                     
    unireedsolomon/cff.c:24610:24: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_type’; did you mean ‘curexc_type’?                        
    24610 |     tmp_type = tstate->exc_type;                                                                                                                    
          |                        ^~~~~~~~                                                                                                                     
          |                        curexc_type                                                                                                                  
    unireedsolomon/cff.c:24611:25: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_value’; did you mean ‘curexc_value’?                      
    24611 |     tmp_value = tstate->exc_value;                                                                                                                  
          |                         ^~~~~~~~~                                                                                                                   
          |                         curexc_value                                                                                                                
    unireedsolomon/cff.c:24612:22: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_traceback’; did you mean ‘curexc_traceback’?              
    24612 |     tmp_tb = tstate->exc_traceback;
          |                      ^~~~~~~~~~~~~
          |                      curexc_traceback
    unireedsolomon/cff.c:24613:13: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_type’; did you mean ‘curexc_type’?
    24613 |     tstate->exc_type = local_type;
          |             ^~~~~~~~
          |             curexc_type
    unireedsolomon/cff.c:24614:13: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_value’; did you mean ‘curexc_value’?
    24614 |     tstate->exc_value = local_value;
          |             ^~~~~~~~~
          |             curexc_value
    unireedsolomon/cff.c:24615:13: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_traceback’; did you mean ‘curexc_traceback’?
    24615 |     tstate->exc_traceback = local_tb;
          |             ^~~~~~~~~~~~~
          |             curexc_traceback
    unireedsolomon/cff.c: In function ‘__Pyx_ExceptionSwap’:
    unireedsolomon/cff.c:24637:24: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_type’; did you mean ‘curexc_type’?
    24637 |     tmp_type = tstate->exc_type;
          |                        ^~~~~~~~
          |                        curexc_type
    unireedsolomon/cff.c:24638:25: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_value’; did you mean ‘curexc_value’?
    24638 |     tmp_value = tstate->exc_value;
          |                         ^~~~~~~~~
          |                         curexc_value
    unireedsolomon/cff.c:24639:22: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_traceback’; did you mean ‘curexc_traceback’?
    24639 |     tmp_tb = tstate->exc_traceback;
          |                      ^~~~~~~~~~~~~
          |                      curexc_traceback
    unireedsolomon/cff.c:24640:13: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_type’; did you mean ‘curexc_type’?
    24640 |     tstate->exc_type = *type;
          |             ^~~~~~~~
          |             curexc_type
    unireedsolomon/cff.c:24641:13: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_value’; did you mean ‘curexc_value’?
    24641 |     tstate->exc_value = *value;
          |             ^~~~~~~~~
          |             curexc_value
    unireedsolomon/cff.c:24642:13: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_traceback’; did you mean ‘curexc_traceback’?
    24642 |     tstate->exc_traceback = *tb;
          |             ^~~~~~~~~~~~~
          |             curexc_traceback
    error: command 'gcc' failed with exit status 1
    ----------------------------------------
```

**Comment by lrq3000** [comment]
Please retry to `pip install --upgrade unireedsolomon`, I should have finally fixed the issue. Thank you very much for your patience! I close this issue meanwhile, please comment if that's not the case and I will reopen!
